catroid/src/main/java/org/catrobat/catroid/cast/CastManager.java

```diff
@@ -0,0 +1,459 @@
+	}
+
+	public synchronized boolean currentlyConnecting() {
+		return (!isConnected && selectedDevice != null);
+	}
+
+	public synchronized void openDeviceSelectorOrDisconnectDialog(Activity activity) {
+		SelectCastDialog dialog = new SelectCastDialog(deviceAdapter, activity);
+		dialog.openDialog();
+	}
+
+	public void setCastButtonFromBaseActivity() {
+		setCastButton(this.castButton);
+	}
+
+	public synchronized void setCastButton(MenuItem castButton) {
+		this.castButton = castButton;
+		castButton.setVisible(mediaRouter.isRouteAvailable(mediaRouteSelector, MediaRouter
+				.AVAILABILITY_FLAG_REQUIRE_MATCH));
+		setIsConnected(isConnected);
+	}
+
+	public void selectRoute(MediaRouter.RouteInfo routeInfo) {
+		mediaRouter.selectRoute(routeInfo);
+	}
+
+	public synchronized void setRemoteLayout(RelativeLayout remoteLayout) {
+		this.remoteLayout = remoteLayout;
+	}
+
+	public synchronized void setRemoteLayoutToIdleScreen(Context context) {
+		remoteLayout.removeAllViews();
+		Drawable drawable = ContextCompat.getDrawable(context, R.drawable.idle_screen_1);
+		remoteLayout.setBackground(drawable);
+	}
+
+	@SuppressLint("InflateParams")
+	public synchronized void setRemoteLayoutToPauseScreen(Context context) {
+		if (remoteLayout != null) {
+			if ((pausedView == null) && !pausedScreenShowing) {
+				pausedView = (RelativeLayout) LayoutInflater.from(context)
+						.inflate(R.layout.cast_pause_screen, null);
+				remoteLayout.addView(pausedView);
+				RelativeLayout.LayoutParams layoutParams = (RelativeLayout.LayoutParams) pausedView.getLayoutParams();
+				Project p = ProjectManager.getInstance().getCurrentProject();
+				layoutParams.height = p.getXmlHeader().getVirtualScreenHeight();
+				layoutParams.width = p.getXmlHeader().getVirtualScreenWidth();
+				pausedView.setLayoutParams(layoutParams);
+				pausedView.setBackgroundColor(0x66000000);
+				pausedScreenShowing = true;
+			}
+			pausedView.setVisibility(View.VISIBLE);
+			pausedScreenShowing = true;
+		}
+	}
+
+	public synchronized void resumeRemoteLayoutFromPauseScreen() {
+		if (remoteLayout != null && pausedView != null) {
+			pausedView.setVisibility(View.GONE);
+			pausedScreenShowing = false;
+		}
+	}
+
+	public synchronized void onStageDestroyed() {
+		if (isConnected) {
+			setRemoteLayoutToIdleScreen(initializingActivity);
+		}
+		stageViewDisplayedOnCast = null;
+		pausedView = null;
+		pausedScreenShowing = false;
+	}
+
+	private class MyMediaRouterCallback extends MediaRouter.Callback {
+
+		private long lastConnectionTry;
+
+		@Override
+		public void onRouteAdded(MediaRouter router, MediaRouter.RouteInfo info) {
+			// Add route to list of discovered routes
+			synchronized (this) {
+				for (int i = 0; i < routeInfos.size(); i++) {
+					MediaRouter.RouteInfo routeInfo = routeInfos.get(i);
+					if (routeInfo.equals(info)) {
+						routeInfos.remove(i);
+					}
+				}
+				routeInfos.add(info);
+				if (castButton != null) {
```

Comment:
please think about if you really need this check. only check if it can be null when you get here.